### ISSUE TITLE
Nginx as reverse proxy not rewriting images - Works on second request

### ISSUE THREAD

**Issue by debanjanbasu** [apache/incubator-pagespeed-mod#1635]
This is the Website:

https://ec2-52-62-229-207.ap-southeast-2.compute.amazonaws.com

We are trying to achieve a Reverse Proxy with Caching and mod_pagespeed to optimize the webapge on: https://5lx4xixnt5h5s-site.azurewebsites.net

One the first load after restarting the server or purging the cache, the user is getting served with stale images.

Here is the Config file attached

[nginx.txt](https://github.com/pagespeed/mod_pagespeed/files/1323131/nginx.txt)

The problem is related to the MapProxyDomain I am pretty sure.

Can you guys please help to ensure an effective solution for the reverse proxy?

Here is what I believe is the issue and the fix.
pagespeed Domain https://ec2-52-62-229-207.ap-southeast-2.compute.amazonaws.com;
        # pagespeed Domain https://*-cdn-webapp.azureedge.net;
        # pagespeed Domain https://*-cdn-storage.azureedge.net;
        # pagespeed Domain https://*-site.azurewebsites.net;
        # pagespeed MapOriginDomain https://girlfriend.staging.atlantic.digitalent-labs.link https://*-cdn-webapp.azureedge.net;
        # pagespeed MapOriginDomain https://girlfriend.staging.atlantic.digitalent-labs.link https://*-cdn-storage.azureedge.net;
        # pagespeed MapOriginDomain https://girlfriend.staging.atlantic.digitalent-labs.link https://*-site.azurewebsites.net;
        pagespeed MapRewriteDomain https://ec2-52-62-229-207.ap-southeast-2.compute.amazonaws.com https://*-cdn-webapp.azureedge.net;
        pagespeed MapRewriteDomain https://ec2-52-62-229-207.ap-southeast-2.compute.amazonaws.com https://*-cdn-storage.azureedge.net;
        pagespeed MapRewriteDomain https://ec2-52-62-229-207.ap-southeast-2.compute.amazonaws.com https://*-site.azurewebsites.net;
        # pagespeed MapProxyDomain https://girlfriend.staging.atlantic.digitalent-labs.link/cdnwebapp/ https://5lx4xixnt5h5s-cdn-webapp.azureedge.net/ https://girlfriend.staging.atlantic.digitalent-labs.link/cdnwebapp/;
        # pagespeed MapProxyDomain https://girlfriend.staging.atlantic.digitalent-labs.link/cdnstorage/ https://5lx4xixnt5h5s-cdn-storage.azureedge.net/ https://girlfriend.staging.atlantic.digitalent-labs.link/cdnstorage/;
        # pagespeed MapProxyDomain https://girlfriend.staging.atlantic.digitalent-labs.link/azurewebsite/ https://5lx4xixnt5h5s-site.azurewebsites.net/ https://girlfriend.staging.atlantic.digitalent-labs.link/azurewebsite/;

**Comment by debanjanbasu** [comment]
Use this URL to purge the cache if needed

https://ec2-52-62-229-207.ap-southeast-2.compute.amazonaws.com/pagespeed_admin/

**Comment by jmarantz** [comment]
What do you mean by "stale images"?

**Comment by debanjanbasu** [comment]
![image](https://user-images.githubusercontent.com/10209115/30725628-6ae44d2e-9f89-11e7-8d33-46603a8b3dc8.png)

This is the response on the first request. It seems the files have not been actually cached yet. But the rewrite_domain is already done it's job. On the second request it starts to work perfectly.

Also the images being served are jpeg. However, the cache stores webp, and on second request the correct image is served.

**Comment by debanjanbasu** [comment]
![image](https://user-images.githubusercontent.com/10209115/30725669-b928bac4-9f89-11e7-82b5-c1e3ab2d3d14.png)

Consecutive request

**Comment by jmarantz** [comment]
I wouldn't call that 'stale' I'd call that 'missing' :)

I can see that the URL works once PageSpeed rewrites it.  But it should
also work prior to the rewrite.  On a cold cache, images get fetched &
rewritten in response to the first request.  However URLs are still
rewritten based on the domain-mapping rules if rewrite_domains is enabled.

I have a crazy guess: you have your CDN origin-fetch configured to match
the .pagespeed. URL pattern?

**Comment by debanjanbasu** [comment]
I agree with you. It's simply a wrong URL, and mod_pagespeed is rewriting the url in the first go. How can I ensure that this works? What do I need to change in my nginx config to make this work please?

**Comment by debanjanbasu** [comment]
The CDN is a standard windows Azure CDN. I've tried with couple of other CDN providers and all are giving the same issue.
